src/main/java/com/google/sps/servlets/GroceryServlet.java

```diff
@@ -27,39 +27,108 @@
+    // checks if the given email matches a email in the family
+    if (assignGrocery.equals("") || assignGrocery.equals(null)) {
+        groceryEntity.setProperty("assignEmail", noneAssigned);
+    } else {
+        ArrayList<String> memberEmails = (ArrayList<String>) familyEntity.getProperty("memberEmails");
+        for(int i = 0; i < memberEmails.size(); i++) {
+            if(assignGrocery.equals(memberEmails.get(i))) {
+                groceryEntity.setProperty("assignEmail" , memberEmails.get(i));
+                break;
+            } else if(i == memberEmails.size() - 1 ) {
+                response.sendRedirect("/grocery.html");
+                return;
+            }
+        }
+    }
+    
     // adds item to datastore
     String grocery = request.getParameter("groceryItem");
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    Entity groceryEntity = new Entity("Grocery");
     if( grocery != null && !grocery.equals("")) {
         groceryEntity.setProperty("grocery", grocery);
+        groceryEntity.setProperty(FAMILY_ID, familyID);
         datastore.put(groceryEntity);
-    }
-               
-    doGet(request,response);  
+    }     
     response.sendRedirect("/grocery.html");
   }
-    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-      // creates arraylist and starts query
-      ArrayList<String> groceryList = new ArrayList<String>();
-      Query query = new Query("Grocery");
-      PreparedQuery results = datastore.prepare(query);
-        
-      //loads entities into arraylist to be printed
-      for (Entity entity : results.asIterable()) {
-          String groceryItem = (String) entity.getProperty("grocery");
-        groceryList.add(groceryItem);
-      }
-      Gson gson = new Gson();
-      response.setContentType("application/json;");
-      response.getWriter().println(gson.toJson(groceryList));
+
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    UserService userService = UserServiceFactory.getUserService();
+
+    // checks if user is a part of a family and returns an error if they aren't
+    String userEmail = userService.getCurrentUser().getEmail();
+    Query query = new Query("UserInfo")
+        .setFilter(new Query.FilterPredicate("email", Query.FilterOperator.EQUAL, userEmail));
+    PreparedQuery results = datastore.prepare(query);
+    Entity userInfoEntity = results.asSingleEntity();
+    if (userInfoEntity == null) {
+        response.setContentType("text/html;");
+        response.getWriter().println("You do not belong to a family yet!");
+    return;
+    }
+
+    // creates arraylist and starts query
+    ArrayList<String> groceryList = new ArrayList<String>();
```

Comment:
I would suggest not predefining groceryList here.

Just define it at 116 and assign it at the same time.